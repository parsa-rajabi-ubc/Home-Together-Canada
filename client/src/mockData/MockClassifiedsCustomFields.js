/**
 * @Author:     Alex Qin
 * @Created:    2021.02.21
 *
 * @Description: mock data for custom fields.
 *
 */

export const rentalsMockCustomFields = {
    price: 800,
    numBed: 5,
    numBath: 3,
    furnished: true,
    petFriendly: false,
    smokingFriendly: false,
    pictures: "pictures"
}

export const houseServicesMockCustomFields = {
    rateAndFees: "666 per day",
    pictures: "pictures"
}

export const agenciesMockCustomFields = {
    rateAndFees: "555 per day",
    pictures: "pictures"
}

export const eventsMockCustomFields = {
    rateAndFees: "666 per day",
    eventDateTime: "2020/2/21 11:00 A.M.",
    contactName: "Alex",
    contactNumber:"778-555-5555",
    pictures: "pictures"
}