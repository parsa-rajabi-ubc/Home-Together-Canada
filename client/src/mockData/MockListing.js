/**
 * @Author:     Parsa Rajabi
 * @Created:    2021.02.20
 *
 * @Description: Mock Listing
 *
 */
import {
    BUSINESS_SERVICE_CATEGORIES, MEMBER_SERVICE_CATEGORIES
} from "../createListing/constants/serviceListingCategoriesText";
import {BUSINESS_CLASSIFIEDS_CATEGORIES} from "../createListing/constants/classifiedListingCategoriesText";
import * as faker from 'faker'

export const mockServiceListings = [
    //CO-HOUSING 0
    {
        title: faker.lorem.words(),
        category: BUSINESS_SERVICE_CATEGORIES.CO_HOUSING,
        fullDescription: faker.lorem.paragraph(),
        contactName: faker.name.findName(),
        unitsForSale: faker.random.number(),
        unitsForRent: faker.random.number(),
    },
    // SHARED HOME 1
    {
        title: faker.lorem.words(),
        category: BUSINESS_SERVICE_CATEGORIES.SHARED_HOME_SERVICES,
        fullDescription: faker.lorem.paragraph(),
        pictures: faker.image.cats(),
        ratesAndFees: faker.random.number() + " " + faker.random.words(),
    },
    // SHARED BUSINESS 2
    {
        title: faker.lorem.words(),
        category: BUSINESS_SERVICE_CATEGORIES.SHARED_BUSINESS_SERVICES,
        fullDescription: faker.lorem.paragraph(),
        pictures: faker.image.cats(),
        ratesAndFees: faker.random.number() + " " + faker.random.words(),
    },
    // GOVERNMENT 3
    {
        title: faker.lorem.words(),
        category: BUSINESS_SERVICE_CATEGORIES.GOVERNMENT_SERVICES,
        fullDescription: faker.lorem.paragraph(),
        contactName: faker.name.findName(),
        contactPhoneNumber: faker.phone.phoneNumber(),
    },
    // MEMBER HOME TO SHARE 4
    {
        title: faker.lorem.words(),
        category: MEMBER_SERVICE_CATEGORIES.MEMBER_HOME,
        fullDescription: faker.lorem.paragraph(),

        generalLocation: faker.address.zipCode(),
        monthlyCost: faker.random.number(),
        pictures: faker.image.cats(),
        numBed: faker.random.number(),
        numBath: faker.random.number(),
        petFriendly: faker.random.boolean(),
        smokeFriendly: faker.random.boolean(),
        utilIncluded: faker.random.boolean(),
    }
]

export const mockClassifiedsListings = [
    //RENTALS 0
    {
        title: faker.lorem.words(),
        category: BUSINESS_CLASSIFIEDS_CATEGORIES.RENTALS,
        fullDescription: faker.lorem.paragraph(),

        price: faker.random.number(),
        pictures: faker.image.cats(),
        ratesAndFees: faker.random.number() + " " + faker.random.words(),
        numBed: faker.random.number(),
        numBath: faker.random.number(),
        furnished: faker.random.boolean(),
        petFriendly: faker.random.boolean(),
        smokeFriendly: faker.random.boolean(),
    },
    // HOUSE_YARD 1
    {
        title: faker.lorem.words(),
        category: BUSINESS_CLASSIFIEDS_CATEGORIES.HOUSE_YARD,
        fullDescription: faker.lorem.paragraph(),

        pictures: faker.image.cats(),
        ratesAndFees: faker.random.number() + " " + faker.random.words(),
    },
    // LEGAL_SALES 2
    {
        title: faker.lorem.words(),
        category: BUSINESS_CLASSIFIEDS_CATEGORIES.LEGAL_SALES,
        fullDescription: faker.lorem.paragraph(),

        pictures: faker.image.cats(),
        ratesAndFees: faker.random.number() + " " + faker.random.words(),
    },
    // CLASSES_CLUBS 3
    {
        title: faker.lorem.words(),
        category: BUSINESS_CLASSIFIEDS_CATEGORIES.CLASSES_CLUBS,
        fullDescription: faker.lorem.paragraph(),

        contactName: faker.name.findName(),
        contactPhoneNumber: faker.phone.phoneNumber(),
        pictures: faker.image.cats(),
        ratesAndFees: faker.random.number() + " " + faker.random.words(),
        eventDateTime: faker.date.soon() + " " + faker.time.recent()
    }
]

export const mockBusinessListingResponse = {
    "id": 6,
    "uid": 2,
    "title": "A Local Calligraphy Classes",
    "shortDescription": "Calligraphy for beginners",
    "fullDescription": "Lorem ipsum dolor .",
    "rateAndFees": "March 1 2021, 10:00 AM",
    "contactName": "Betsy",
    "contactPhoneNumber": 4036527153,
    "eventDateTime": 4036527153,
    "isClassified": true,
    "createdAt": "2021-02-23T18:54:05.000Z",
    "updatedAt": "2021-02-23T18:54:05.000Z",
    "categoryName": "Classes, Clubs & Events",
    "business": {
        "username": "apples0",
        "email": "apples0@gelden.com",
        "uid": 2,
        "businessName": "Bob's Bricks",
        "logo": null,
        "isIncorporated": false,
        "incorporatedOwnersNames": null,
        "businessPhoneNumber": "333-333-3333",
        "businessCellPhoneNumber": "444-444-4444",
        "isNationWide": true,
        "mapAddressLine1": null,
        "mapAddressLine2": null,
        "mapCity": null,
        "mapProvince": null,
        "mapPostalCode": null,
        "mapLatitude": "36.0195234",
        "mapLongitude": "14.2385650",
        "website": "www.google.com",
        "createdAt": "2021-02-23T18:54:01.000Z",
        "updatedAt": "2021-02-23T18:54:01.000Z"
    }
}

export const mockMemberListingResponse = {
    "id": 18,
    "uid": 5,
    "title": "Empty Room in Condo",
    "shortDescription": "Looking for an easy going roommate",
    "fullDescription": "Lorem ipsum .",
    "monthlyCost": 600,
    "numBed": 2,
    "numBath": 1,
    "petFriendly": true,
    "smokeFriendly": false,
    "postalCode": "T2V1S5",
    "utilitiesIncluded": true,
    "isClassified": false,
    "createdAt": "2021-02-24T01:31:58.000Z",
    "updatedAt": "2021-02-24T01:31:58.000Z",
    "categoryName": "Members with Homes to Share"
}
