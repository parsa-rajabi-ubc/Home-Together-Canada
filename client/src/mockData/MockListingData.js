/**
 * @Author:     Jeff Hatton
 * @Created:    2020.02.16
 *
 * @Description: Const array of mock listings - includes listings for members and businesses
 *
 */

export const mockMemberListings = [
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, Utilities Covered",
        monthlyCost: 750,
        petFriendly: false,
        smokeFriendly: false,
        datePosted: "2021-02-14T02:42:39.005Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Unfurnished Room, Utilities/internet Split",
        monthlyCost: 500,
        petFriendly: true,
        smokeFriendly: true,
        datePosted: "2021-02-15T05:42:39.000Z"
    },
    {
        title: "Basement suite",
        shortDescription: "1 bed 1 bath, nice part of town, parking spot",
        monthlyCost: 1100,
        petFriendly: false,
        smokeFriendly: false,
        datePosted: "2021-02-14T16:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, Utilities Covered",
        monthlyCost: 850,
        petFriendly: false,
        smokeFriendly: true,
        datePosted: "2021-02-16T23:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, Utilities Covered, Private Kitchenette",
        monthlyCost: 950,
        petFriendly: true,
        smokeFriendly: true,
        datePosted: "2021-02-16T03:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Unfurnished Room, spacious living area ",
        monthlyCost: 775,
        petFriendly: false,
        smokeFriendly: false,
        datePosted: "2021-02-15T11:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Unfurnished Room, large kitchen storage",
        monthlyCost: 750,
        petFriendly: true,
        smokeFriendly: false,
        datePosted: "2021-02-13T18:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, significant storage space",
        monthlyCost: 650,
        petFriendly: false,
        smokeFriendly: false,
        datePosted: "2021-02-13T05:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, private bathroom",
        monthlyCost: 750,
        petFriendly: true,
        smokeFriendly: false,
        datePosted: "2021-02-15T05:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, fast internet",
        monthlyCost: 850,
        petFriendly: false,
        smokeFriendly: false,
        datePosted: "2021-02-16T01:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished, private bath",
        monthlyCost: 950,
        petFriendly: true,
        smokeFriendly: false,
        datePosted: "2021-02-16T02:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, Utilities Covered",
        monthlyCost: 750,
        petFriendly: true,
        smokeFriendly: true,
        datePosted: "2021-02-14T08:42:39.000Z"
    },
    {
        title: "Rooms for Rent",
        shortDescription: "3 Furnished Rooms, for students preferably",
        monthlyCost: 650,
        petFriendly: false,
        smokeFriendly: true,
        datePosted: "2021-02-14T07:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, beautiful house",
        monthlyCost: 800,
        petFriendly: true,
        smokeFriendly: false,
        datePosted: "2021-02-15T05:42:39.000Z"
    },
    {
        title: "Basement suite for Couple",
        shortDescription: "Furnished Large Room, Utilities Covered",
        monthlyCost: 1300,
        petFriendly: true,
        smokeFriendly: true,
        datePosted: "2021-02-13T12:42:39.000Z"
    }
];

export const mockBusinessListings = [
    {
        logo: "public/uploads/logo1",
        title: "Apartments for 2 available",
        businessName: "Kelowna Apartment Inc.",
        shortDescription: "2 bed 2 bath - furnished",
        datePosted: "2021-02-14T22:42:39.000Z"
    },
    {
        logo: "public/uploads/logo2",
        title: "Apartments for 3 available",
        businessName: "Kelowna Apartment Inc.",
        shortDescription: "3 bed 2 bath - furnished",
        datePosted: "2021-02-15T05:42:39.000Z"
    },
    {
        logo: "public/uploads/logo3",
        title: "House with 4 rooms available",
        businessName: "Kelowna House Rentals",
        shortDescription: "4 bed 3 bath - furnished",
        datePosted: "2021-02-14T16:42:39.000Z"
    },
    {
        logo: "public/uploads/logo4",
        title: "Beautiful House On Lakefront For Rent",
        businessName: "Lakefront Properties",
        shortDescription: "2 bed 2 bath - furnished",
        datePosted: "2021-02-16T23:42:39.000Z"
    },
    {
        logo: "public/uploads/logo5",
        title: "Vacuum Rentals",
        businessName: "Kelowna Apartment Inc.",
        shortDescription: "2 bed 2 bath - furnished",
        datePosted: "2021-02-16T03:42:39.000Z"
    },
    {
        logo: "public/uploads/logo6",
        title: "Lawn Management Services",
        businessName: "Lawns-R-Us",
        shortDescription: "2 bed 2 bath - furnished",
        datePosted: "2021-02-15T11:42:39.000Z"
    },
    {
        logo: "public/uploads/logo7",
        title: "Carpet Cleaning Rentals",
        businessName: "Carpet Cleaners",
        shortDescription: "We offer services, or tool rentals",
        datePosted: "2021-02-13T18:42:39.000Z"
    },
    {
        logo: "public/uploads/logo8",
        title: "Plumbing Services Available",
        businessName: "Kelowna Plumbing Inc.",
        shortDescription: "2 bed 2 bath - furnished",
        datePosted: "2021-02-13T05:42:39.000Z"
    },
    {
        logo: "public/uploads/logo9",
        title: "Apartment Rooms Available",
        businessName: "Kelowna Apartment Inc.",
        shortDescription: "Apartments of various sizes",
        datePosted: "2021-02-15T05:42:39.000Z"
    },
    {
        logo: "public/uploads/logo10",
        title: "Window Cleaning Services",
        businessName: "Kelowna Glass Cleaners",
        shortDescription: "We clean residential or high rise complexes",
        datePosted: "2021-02-16T01:42:39.000Z"
    },
    {
        logo: "public/uploads/logo11",
        title: "Apartments for 2 available",
        businessName: "Kelowna Apartment Inc.",
        shortDescription: "2 bed 1 bath - unfurnished",
        datePosted: "2021-02-16T02:42:39.000Z"
    },
    {
        logo: "public/uploads/logo12",
        title: "Apartments for 2 available",
        businessName: "Kelowna Apartment Inc.",
        shortDescription: "2 bed 2 bath - Unfurnished",
        datePosted: "2021-02-14T08:42:39.000Z"
    },
    {
        logo: "public/uploads/logo13",
        title: "Apartments for 3 available",
        businessName: "Kelowna Apartment Inc.",
        shortDescription: "3 bed 2 bath - Unfurnished",
        datePosted: "2021-02-14T07:42:39.000Z"
    },
    {
        logo: "public/uploads/logo14",
        title: "Senior services",
        businessName: "Senior services",
        shortDescription: "Mobility tools, awareness of area services, financial aid",
        datePosted: "2021-02-15T05:42:39.000Z"
    },
    {
        logo: "public/uploads/logo15",
        title: "Shared Vehicle Insurance",
        businessName: "Kelowna Vehicle Insurance",
        shortDescription: "Insurance and help available for groups sharing cars",
        datePosted: "2021-02-13T12:42:39.000Z"
    }
];