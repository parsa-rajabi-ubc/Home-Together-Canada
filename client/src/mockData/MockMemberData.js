/**
 * @Author:     Parsa Rajabi
 * @Created:    June 2022
 *
 * @Description: Mock Member data for automated testing
 *
 */
const {faker} = require("@faker-js/faker");

const MOCK_MEMBER_DATA = {
    firstName: faker.name.firstName(),
    lastName: faker.name.lastName(),
    email: 'tech@hometogether.ca',
    phoneNumber: {
        areaCode: faker.random.numeric(3),
        prefix: faker.random.numeric(3),
        suffix: faker.random.numeric(4)
    },
    address: {
        street: faker.address.streetAddress(),
        apt: faker.address.buildingNumber(),
        city: faker.address.city(),
        province: "Alberta",
        postCode: "T3T1T3"
    },
    monthlyRent: {
        min: Math.floor(Math.random() * 999) + 500,
        max: Math.floor(Math.random() * 2500) + 1000,
    },
    preferences:{
        petFriendly: faker.datatype.boolean(),
        smoking: faker.datatype.boolean(),
        mobile: faker.datatype.boolean(),
        allergies: faker.datatype.boolean(),
        religion: faker.datatype.boolean(),
        diet: faker.datatype.boolean(),
        hasHome: faker.datatype.boolean(),
        interestInBuyingHome: faker.datatype.boolean(),
    },
    aboutSelf: faker.random.words(2),

    sharingPreferences:{
        genders: ["Male", "Female", "Other"],
        age:{
            min: Math.floor(Math.random() * 29) + 18,
            max: Math.floor(Math.random() * 60) + 30,
        },
        overlapBudget:{
            min: Math.floor(Math.random() * 999) + 500,
            max: Math.floor(Math.random() * 2500) + 1000,
        }
    },
    accountDetails:{
        username: "TestAccount"+faker.random.numeric(2),
        password: "Bose1234"
    }

}

module.exports = {MOCK_MEMBER_DATA}