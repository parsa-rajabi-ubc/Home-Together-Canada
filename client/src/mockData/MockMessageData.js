/**
 * @Author:     Jeff Hatton
 * @Created:    2020.03.03
 *
 * @Description: Const array of mock messages - includes multiple conversations, most between messageMember and other members with differing timestamps
 *
 */

export const mockMessages = [
    {
        id: 1,
        messageContent: "message content 1 all on different days and times",
        senderId: 1,
        senderUsername: "member1",
        receiverId: 2,
        receiverUsername: "member2",
        dateSent: "2021-01-12T02:42:39.005Z"
    },
    {
        id: 2,
        messageContent: "message content 2",
        senderId: 1,
        senderUsername: "member1",
        receiverId: 2,
        receiverUsername: "member2",
        dateSent: "2021-01-01T02:42:39.005Z"
    },
    {
        id: 3,
        messageContent: "message content 3",
        senderId: 1,
        senderUsername: "member1",
        receiverId: 2,
        receiverUsername: "member2",
        dateSent: "2021-02-14T08:42:39.005Z"
    },
    {
        id: 4,
        messageContent: "message content 4",
        senderId: 1,
        senderUsername: "member1",
        receiverId: 2,
        receiverUsername: "member2",
        dateSent: "2021-01-12T02:42:39.005Z"
    },
    {
        id: 5,
        messageContent: "message content 5",
        senderId: 2,
        senderUsername: "member2",
        receiverId: 1,
        receiverUsername: "member1",
        dateSent: "2020-03-13T02:42:39.075Z"
    },
    {
        id: 6,
        messageContent: "message content 6",
        senderId: 1,
        senderUsername: "member1",
        receiverId: 2,
        receiverUsername: "member2",
        dateSent: "2021-02-17T02:43:39.005Z"
    },
    {
        id: 7,
        messageContent: "message content 7",
        senderId: 2,
        senderUsername: "member2",
        receiverId: 1,
        receiverUsername: "member1",
        dateSent: "2021-01-19T09:42:39.005Z"
    },
    {
        id: 8,
        messageContent: "message content 8",
        senderId: 2,
        senderUsername: "member2",
        receiverId: 1,
        receiverUsername: "member1",
        dateSent: "2021-01-11T02:42:39.005Z"
    },
    {
        id: 9,
        messageContent: "message content 9 has timestamp before a smaller message id",
        senderId: 1,
        senderUsername: "member1",
        receiverId: 2,
        receiverUsername: "member2",
        dateSent: "2021-01-10T02:42:39.005Z"
    },
    {
        id: 10,
        messageContent: "message content 10",
        senderId: 2,
        senderUsername: "member2",
        receiverId: 1,
        receiverUsername: "member1",
        dateSent: "2021-02-10T02:42:39.005Z"
    }
];