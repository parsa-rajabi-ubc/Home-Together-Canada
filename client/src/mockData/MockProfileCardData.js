/**
 * @Author:     Jeff Hatton
 * @Created:    2020.01.12
 *
 * @Description: Const array of mock profiles
 *
 */

const mockProfiles = [
    {
        username: "AliceinWonderLand",
        gender: "Female",
        birthYear: 1985,
        status: "Single",
        minMonthlyBudget: 200,
        maxMonthlyBudget: 550,
        hasPets: false,
        isSmoker: false,
        isReligionImportant: false,
        isDietImportant: false,
        hasHomeToShare: false
    },
    {
        username: "Babar",
        gender: "Male",
        birthYear: 1990,
        status: "Couple With Children",
        minMonthlyBudget: 2000,
        maxMonthlyBudget: 5500,
        hasPets: true,
        isSmoker: true,
        isReligionImportant: true,
        isDietImportant: true,
        hasHomeToShare: true
    },
    {
        username: "CREATINE",
        gender: "Other",
        birthYear: 1995,
        status: "Single",
        minMonthlyBudget: 800,
        maxMonthlyBudget: 1200,
        hasPets: false,
        isSmoker: true,
        isReligionImportant: true,
        isDietImportant: true,
        hasHomeToShare: true
    },
    {
        username: "DonaldDuck",
        gender: "Male",
        birthYear: 1970,
        status: "Single Parent",
        minMonthlyBudget: 500,
        maxMonthlyBudget: 600,
        hasPets: false,
        isSmoker: false,
        isReligionImportant: true,
        isDietImportant: true,
        hasHomeToShare: true
    },
    {
        username: "EusticeOfNarnia",
        gender: "Male",
        birthYear: 1950,
        status: "Single",
        minMonthlyBudget: 1300,
        maxMonthlyBudget: 2300,
        hasPets: false,
        isSmoker: false,
        isReligionImportant: false,
        isDietImportant: true,
        hasHomeToShare: false
    },
    {
        username: "Foxtrot",
        gender: "Other",
        birthYear: 1980,
        status: "Couple",
        minMonthlyBudget: 1400,
        maxMonthlyBudget: 1600,
        hasPets: false,
        isSmoker: false,
        isReligionImportant: true,
        isDietImportant: false,
        hasHomeToShare: true
    },
    {
        username: "Grinch",
        gender: "Other",
        birthYear: 1960,
        status: "Single",
        minMonthlyBudget: 2500,
        maxMonthlyBudget: 3700,
        hasPets: false,
        isSmoker: true,
        isReligionImportant: false,
        isDietImportant: false,
        hasHomeToShare: false
    },
    {
        username: "Hydra",
        gender: "Female",
        birthYear: 1975,
        status: "Couple With Children",
        minMonthlyBudget: 1200,
        maxMonthlyBudget: 2550,
        hasPets: true,
        isSmoker: false,
        isReligionImportant: false,
        isDietImportant: false,
        hasHomeToShare: false
    },
    {
        username: "InnigoMontoya",
        gender: "Male",
        birthYear: 1970,
        status: "Single Parent",
        minMonthlyBudget: 300,
        maxMonthlyBudget: 575,
        hasPets: false,
        isSmoker: true,
        isReligionImportant: false,
        isDietImportant: false,
        hasHomeToShare: false
    },
    {
        username: "JamesBond",
        gender: "Male",
        birthYear: 1950,
        status: "Single",
        minMonthlyBudget: 200,
        maxMonthlyBudget: 550,
        hasPets: false,
        isSmoker: true,
        isReligionImportant: false,
        isDietImportant: true,
        hasHomeToShare: false
    },
    {
        username: "Kirby",
        gender: "Other",
        birthYear: 1976,
        status: "Existing Group",
        minMonthlyBudget: 200,
        maxMonthlyBudget: 2550,
        hasPets: false,
        isSmoker: false,
        isReligionImportant: false,
        isDietImportant: false,
        hasHomeToShare: false
    },
    {
        username: "Link",
        gender: "Male",
        birthYear: 2001,
        status: "Couple",
        minMonthlyBudget: 500,
        maxMonthlyBudget: 700,
        hasPets: true,
        isSmoker: true,
        isReligionImportant: false,
        isDietImportant: false,
        hasHomeToShare: true
    },
    {
        username: "Mario",
        gender: "Male",
        birthYear: 1950,
        status: "Single",
        minMonthlyBudget: 900,
        maxMonthlyBudget: 2600,
        hasPets: false,
        isSmoker: true,
        isReligionImportant: true,
        isDietImportant: false,
        hasHomeToShare: true
    },
    {
        username: "Nero",
        gender: "Male",
        birthYear: 1978,
        status: "Existing Group",
        minMonthlyBudget: 3300,
        maxMonthlyBudget: 6650,
        hasPets: false,
        isSmoker: true,
        isReligionImportant: false,
        isDietImportant: false,
        hasHomeToShare: true
    },
    {
        username: "OddDuck",
        gender: "Female",
        birthYear: 1998,
        status: "Single",
        minMonthlyBudget: 400,
        maxMonthlyBudget: 1950,
        hasPets: false,
        isSmoker: true,
        isReligionImportant: true,
        isDietImportant: true,
        hasHomeToShare: true
    },
    {
        username: "PlatoInPerson",
        gender: "Male",
        birthYear: 1950,
        status: "Single Parent",
        minMonthlyBudget: 100,
        maxMonthlyBudget: 650,
        hasPets: true,
        isSmoker: false,
        isReligionImportant: false,
        isDietImportant: true,
        hasHomeToShare: false
    },
    {
        username: "QWERTYworker",
        gender: "Male",
        birthYear: 1987,
        status: "Single",
        minMonthlyBudget: 2200,
        maxMonthlyBudget: 3550,
        hasPets: false,
        isSmoker: true,
        isReligionImportant: false,
        isDietImportant: false,
        hasHomeToShare: true
    },
    {
        username: "Rattata",
        gender: "Female",
        birthYear: 1988,
        status: "Single",
        minMonthlyBudget: 700,
        maxMonthlyBudget: 1250,
        hasPets: false,
        isSmoker: true,
        isReligionImportant: false,
        isDietImportant: false,
        hasHomeToShare: false
    },
    {
        username: "Sierra",
        gender: "Female",
        birthYear: 1945,
        status: "Single",
        minMonthlyBudget: 800,
        maxMonthlyBudget: 1150,
        hasPets: false,
        isSmoker: true,
        isReligionImportant: false,
        isDietImportant: false,
        hasHomeToShare: true
    },
    {
        username: "TooFriendly",
        gender: "Male",
        birthYear: 1970,
        status: "Single",
        minMonthlyBudget: 700,
        maxMonthlyBudget: 1050,
        hasPets: false,
        isSmoker: true,
        isReligionImportant: false,
        isDietImportant: false,
        hasHomeToShare: false
    },
    {
        username: "UltimateRoomie123",
        gender: "Other",
        birthYear: 1996,
        status: "Single",
        minMonthlyBudget: 2400,
        maxMonthlyBudget: 2550,
        hasPets: false,
        isSmoker: true,
        isReligionImportant: false,
        isDietImportant: false,
        hasHomeToShare: true
    },
    {
        username: "Ventriloquist45",
        gender: "Male",
        birthYear: 1970,
        status: "Existing Group",
        minMonthlyBudget: 900,
        maxMonthlyBudget: 2050,
        hasPets: false,
        isSmoker: true,
        isReligionImportant: false,
        isDietImportant: false,
        hasHomeToShare: true
    },
    {
        username: "Wario",
        gender: "Male",
        birthYear: 1991,
        status: "Single",
        minMonthlyBudget: 700,
        maxMonthlyBudget: 1350,
        hasPets: false,
        isSmoker: true,
        isReligionImportant: false,
        isDietImportant: false,
        hasHomeToShare: true
    },
    {
        username: "XenaThePrincess",
        gender: "Female",
        birthYear: 1981,
        status: "Single",
        minMonthlyBudget: 1100,
        maxMonthlyBudget: 3550,
        hasPets: false,
        isSmoker: true,
        isReligionImportant: false,
        isDietImportant: false,
        hasHomeToShare: false
    },
    {
        username: "YogurtFanatic",
        gender: "Male",
        birthYear: 1991,
        status: "Couple With Children",
        minMonthlyBudget: 1450,
        maxMonthlyBudget: 1550,
        hasPets: false,
        isSmoker: true,
        isReligionImportant: false,
        isDietImportant: false,
        hasHomeToShare: true
    },
    {
        username: "Zelda995",
        gender: "Female",
        birthYear: 1995,
        status: "Couple",
        minMonthlyBudget: 700,
        maxMonthlyBudget: 980,
        hasPets: false,
        isSmoker: true,
        isReligionImportant: false,
        isDietImportant: false,
        hasHomeToShare: false
    }
]

export default mockProfiles;
