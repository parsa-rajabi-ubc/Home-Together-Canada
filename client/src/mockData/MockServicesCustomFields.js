/**
 * @Author:     Alex Qin
 * @Created:    2021.02.18
 *
 * @Description: mock data for custom fields.
 *
 */


export const memberMockCustomFields = {
    streetLine1: "123 Candy Lane",
    streetLine2: "",
    city: "Kelowna",
    province: "BC",
    postalCode: "V1X 1X3",
    homeShareMonthlyCost: 800,
    numBed: 5,
    numBath: 3,
    utilIncluded: true,
    petFriendly: false,
    smokeFriendly: false,
    pictures: "pictures"
}

export const cohousingMockCustomFields = {
    contactName: "Alex",
    unitsForSale: 5,
    unitsForRent: 5,
}

export const homeServicesMockCustomFields = {
    rateAndFees: "666 per day",
    pictures: "pictures"
}

export const governmentMockCustomFields = {
    contactPerson: "Alex",
    phoneNumber:"778-555-5555"
}