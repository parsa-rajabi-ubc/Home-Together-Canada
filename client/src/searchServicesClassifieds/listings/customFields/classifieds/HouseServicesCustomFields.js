/**
 * @Author:     Alex Qin
 * @Created:    2021.02.21
 *
 * @Description: Re-usable House Services custom fields.
 *
 */
import PropTypes from "prop-types";
import React from 'react';
import {ImageGallery} from "../../../../common/listings/ImageGallery";

const HOUSE_SERVICES_CUSTOM_FIELDS_TEXT = {
    RATE_AND_FEES: "Rates and fees (CAD)",
    PICTURES: "Pictures"
}

const HouseServicesCustomFields = (props) => {
    const { rateAndFees, pictures} = props;

    return (
        <div>
            <label className={"label-result"}>{HOUSE_SERVICES_CUSTOM_FIELDS_TEXT.RATE_AND_FEES}</label>
            <p> {rateAndFees}</p>

            {!!pictures.length &&
                <div>
                    <label className={"label-result"}>{HOUSE_SERVICES_CUSTOM_FIELDS_TEXT.PICTURES}</label>
                    <ImageGallery pictures={pictures}/>
                </div>
            }
        </div>
    );
}

HouseServicesCustomFields.propTypes = {
    rateAndFees: PropTypes.string.isRequired,
    pictures: PropTypes.arrayOf(PropTypes.string),
}

export default HouseServicesCustomFields;