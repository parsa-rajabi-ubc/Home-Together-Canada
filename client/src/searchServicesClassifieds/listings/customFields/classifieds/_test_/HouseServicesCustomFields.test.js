/**
 * @Author:     Alex Qin
 * @Created:    2021.02.21
 *
 * @Description: HouseServicesCustomFields snapshot test. Returns a snapshot.
 *
 */
import React from 'react';
import renderer from  'react-test-renderer'
import {houseServicesMockCustomFields} from "../../../../../mockData/MockClassifiedsCustomFields";
import HouseServicesCustomFields from "../HouseServicesCustomFields";

describe('HouseServicesCustomFields', () => {
    it("should render correctly regardless of properties", () => {
        // given
        const props = {
            ...houseServicesMockCustomFields,
        };
        //when
        const component = renderer.create(<HouseServicesCustomFields {...props}/>);
        const tree = component.toJSON();
        //then
        expect(tree).toMatchSnapshot();
    });
});