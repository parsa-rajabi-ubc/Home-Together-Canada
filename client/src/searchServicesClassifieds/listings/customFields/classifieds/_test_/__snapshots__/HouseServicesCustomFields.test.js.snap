// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HouseServicesCustomFields should render correctly regardless of properties 1`] = `
<div>
  <label
    className="label-result"
  >
    Rates and fees (CAD)
  </label>
  <p>
     
    666 per day
  </p>
</div>
`;
